### PR TITLE
Write Python Analytics Documentation

### PR DESCRIPTION
**Pull request by arthurp**

### CODE REVIEW

**Comment by gurbinder533** [comment]
Looks good.

**Comment by arthurp** [comment]
@gurbinder533 This is ready. I'm going to wrap and document `louvain_clustering` in a separate PR.

File: libgalois/include/katana/PropertyGraph.h [hunk 1]
```diff
@@ -493,12 +490,8 @@ KATANA_EXPORT GraphTopology::Edge FindEdgeSortedByDest(
...
     GraphTopology::Node node_to_find);
 
-/// SortNodesByDegree relables node ids by sorting in the descending
-/// order by node degree
-///
-/// This function modifies the PropertyGraph topology by in-place
-/// relabeling and sorting the node ids by their degree in the
-/// descending order.
+/// Relabel all nodes in the graph by sorting in the descending
+//  order by node degree.
```
Comment (gurbinder533) [inline libgalois/include/katana/PropertyGraph.h]: ///

File: libgalois/include/katana/analytics/Plan.h [hunk 2]
```diff
@@ -12,13 +12,27 @@ enum Architecture {
   kDistributed
 };
 
+/// The base class for abstract algorithm execution plans.
+///
+/// Execution plans contain any tuning parameters the abstract algorithm requires. In general, this will include
+/// selecting the concrete algorithm to use and providing it's tuning parameters. Plans do not affect the result (beyond
+/// potentially returning a different valid result in the case of non-deterministic algorithms). For example, the tile
```
Comment (gurbinder533) [inline libgalois/include/katana/analytics/Plan.h]: We don't expect the tile size to result in different results for all the algorithms. I would just say "..case of non-deterministic algorithm or floating-point operations"

File: libgalois/include/katana/analytics/Plan.h [hunk 3]
```diff
@@ -12,13 +12,27 @@ enum Architecture {
...
+/// size of an algorithm.
+///
+/// All plans provide static methods to create a plan each concrete algorithm and constructors for any automatic
+/// heuristics. The automatic constructors may either take no input and make decisions at run time, use fixed default
+/// parameters, or take a graph to be analyzed to determine appropriate paramters. A plan created automatically for one
```
Comment (gurbinder533) [inline libgalois/include/katana/analytics/Plan.h]: parameters

File: python/katana/analytics/_connected_components.pyx [hunk 4]
```diff
@@ -130,50 +164,107 @@ cdef class ConnectedComponentsPlan(Plan):
...
+        """
+        Serial connected components algorithm. Uses union-find data structure.
+        """
         return ConnectedComponentsPlan.make(_ConnectedComponentsPlan.Serial())
     @staticmethod
     def label_prop() -> ConnectedComponentsPlan:
+        """
+        Label propagation push-style algorithm. Initially, all nodes are in
+        their own component ids (same as their node ids). Then, the component
+        ids are set the minimum component id in one's neighborhood.
```
Comment (gurbinder533) [inline python/katana/analytics/_connected_components.pyx]: "are set to the ...."

File: python/katana/analytics/plan.pyx [hunk 5]
```diff
@@ -2,14 +2,63 @@ from enum import Enum
...
     Distributed = _Architecture.kDistributed
 
 
 cdef class Plan:
+    """
+    Each abstract algorithm has an execution `Plan` subclass.
+
+    Execution plans contain any tuning parameters the abstract algorithm requires. In general, this will include
+    selecting the concrete algorithm to use and providing it's tuning parameters. Plans do not affect the result (beyond
+    potentially returning a different valid result in the case of non-deterministic algorithms). For example, the tile
```
Comment (gurbinder533) [inline python/katana/analytics/plan.pyx]: Same as above.

**Review by gurbinder533 — APPROVED** [review]
LGTM. Just a few minor comments.

File: libgalois/include/katana/analytics/Plan.h (hunk 2 above)
Comment (arthurp) [reply libgalois/include/katana/analytics/Plan.h]: This paragraph doesn't imply that the tile size will change the results to me. I'm not sure what you are responding to.

Comment (arthurp) [reply libgalois/include/katana/analytics/Plan.h]: I read more carefully and I understand. Fixed.

**Pull request merged**